action.yml

```diff
@@ -12,9 +12,13 @@ inputs:
     description: 'Version of ESP-IDF docker image to use'
     default: 'latest'
     required: false
+  esp_target:
+    description: 'ESP32 variant to build for'
+    default: 'esp32'
+    required: false
 
 runs:
   using: "composite"
   steps:
-    - run: docker run -t -v "${GITHUB_WORKSPACE}/${{ inputs.path }}:/app" -w=/app espressif/idf:${{ inputs.esp_idf_version }} idf.py build
+    - run: docker run -t -e IDF_TARGET="${{ inputs.esp_target }}" -v "${GITHUB_WORKSPACE}/${{ inputs.path }}:/app" -w=/app espressif/idf:${{ inputs.esp_idf_version }} idf.py build
```

Comment:
Unfortunately calling `set-target` is a non-starter for my use case as it will delete any pre-existing `sdkconfig` file which I'm currently using to seed a config option of my project. I think it would be better to have ESP-IDF accept the flexibility in values on `IDF_TARGET` as well.

Reply:
@atanisoft changing behavior of `IDF_TARGET` env variable for all versions of IDF  is impossible, but we can handle it right here:

```
- run: |
  export IDF_TARGET=$(echo "${{ inputs.target }}" | tr '[:upper:]' '[:lower:]' | tr -d '_-') 
  docker run -t -e IDF_TARGET="${IDF_TARGET}" -v "${GITHUB_WORKSPACE}/${{ inputs.path }}:/app" -w=/app espressif/idf:${{ inputs.esp_idf_version }} idf.py build
```

Reply:
I agree fixing all versions of IDF is not feasible, but fixing it for supported versions should be doable.

Your proposal looks like it should work, let me update it and push the update.